modules/pubxaiAnalyticsAdapter.js

```diff
@@ -1,208 +1,429 @@
+        'consentDetail',
+        'pmacDetail',
+        'initOptions',
+      ],
+      eventType: 'win',
+    },
+    {
+      path: auctionPath,
+      requiredKeys: [
+        'bids',
+        'pageDetail',
+        'deviceDetail',
+        'floorDetail',
+        'auctionDetail',
+        'userDetail',
+        'consentDetail',
+        'pmacDetail',
+        'initOptions',
+      ],
+      eventType: 'auction',
+    },
+  ].forEach(({ path, requiredKeys, eventType }) => {
+    const data = Object.fromEntries(
+      requiredKeys.map((key) => [key, auctionData[key]])
+    );
+    if (
+      auctionCache[auctionId].sendAs.includes(eventType) ||
+      !requiredKeys.every((key) => !!auctionData[key])
+    ) {
+      return;
     }
-    data.deviceDetail = {};
-    Object.assign(data.deviceDetail, {
-      platform: navigator.platform,
-      deviceType: getDeviceType(),
-      deviceOS: getOS(),
-      browser: getBrowser()
-    });
-
-    let pubxaiAnalyticsRequestUrl = buildUrl({
+    const pubxaiAnalyticsRequestUrl = buildUrl({
       protocol: 'https',
-      hostname: (initOptions && initOptions.hostName) || defaultHost,
-      pathname: status == 'bidwon' ? winningBidPath : auctionPath,
+      hostname:
+        (auctionData.initOptions && auctionData.initOptions.hostName) ||
+        defaultHost,
+      pathname: path,
       search: {
-        auctionTimestamp: auctionTimestamp,
+        auctionTimestamp: auctionData.auctionDetail.timestamp,
         pubxaiAnalyticsVersion: pubxaiAnalyticsVersion,
-        prebidVersion: getGlobal().version
+        prebidVersion: getGlobal().version,
+      },
+    });
+    sendCache[pubxaiAnalyticsRequestUrl].push(data);
+    auctionCache[auctionId].sendAs.push(eventType);
+  });
+};
+
+const send = () => {
+  const toBlob = (d) => new Blob([JSON.stringify(d)], { type: 'text/json' });
+
+  Object.entries(sendCache).forEach(([requestUrl, events]) => {
+    let payloadStart = 0;
+
+    events.forEach((event, index, arr) => {
+      const payload = arr.slice(payloadStart, index + 2);
+      const payloadTooLarge = toBlob(payload).size > 65536;
+
+      if (payloadTooLarge || index + 1 === arr.length) {
+        navigator.sendBeacon(
+          requestUrl,
+          toBlob(payloadTooLarge ? payload.slice(0, -1) : payload)
+        );
+        payloadStart = index;
       }
     });
-    if (status == 'bidwon') {
-      ajax(pubxaiAnalyticsRequestUrl, undefined, JSON.stringify(data), { method: 'POST', contentType: 'text/json' });
-    } else if (status == 'auctionEnd' && auctionCache.indexOf(data.initOptions.auctionId) === -1) {
-      ajax(pubxaiAnalyticsRequestUrl, undefined, JSON.stringify(data), { method: 'POST', contentType: 'text/json' });
-      auctionCache.push(data.initOptions.auctionId);
+
+    events.splice(0);
+  });
+};
+
+// register event listener to send logs when user leaves page
+if (hasSendBeaconSupport()) {
+  document.addEventListener('visibilitychange', () => {
```

Comment:
Just use fetch keepalive

Reply:
`keepalive` is not supported in all the browsers. I'm unsure if switching to keepalive is a good idea.

Reply:
if you want to use sendBeacon, you'll need to commit that to core communication methods

It isn't in fox but well be imminently is my understanding.

Reply:
some activity here https://bugzilla.mozilla.org/show_bug.cgi?id=1342484 just two days ago

Reply:
https://github.com/prebid/Prebid.js/blob/2efc834cc291b3c2ff9abc70e95d5b3d3eb01b63/modules/genericAnalyticsAdapter.js#L145

It's the simplest way to implement